### DISCUSSION TITLE
Parse SQL and for each column get column name, table name, column alias and table alias

### DISCUSSION THREAD

**Discussion by Vipin-Sharma** [JSQLParser/JSqlParser#1717] [Q&A]
What is best way to get column name, table name, column alias and table alias for each column in a SQL?

As given on the question https://stackoverflow.com/questions/58984896/listing-all-the-tables-columns-of-parsed-sql-select-statement I understand JSQLParser is only parser, we need information of DB schema as well in order to get this information. I have DDL file so I can read info related to schema. What would be best way to do this with help of JSQLParser? If can not be done completely via JSQLParser then what all things JSQLParser can do in this case and if you have suggestion for next steps that would be very helpful.

**Comment by manticore-projects** [comment]
Greetings.

Step 1) use JSQLParser to get all the involved tables
Step 2) use JDBC MetaData to return the available columns for those tables
Step 3) use JSQLParser to get all Column objects
Step 4) map the columns to the available columns

This will work for the very most statements although it may be possible to craft some evil statements, where it may deliver wrong/incomplete results.

Good luck!
